ambry-store/src/test/java/com.github.ambry.store/LogSegmentTest.java

```diff
@@ -247,11 +247,11 @@ public void readTest() throws IOException {
         }
       }
 
-      // position + buffer.remaining() > endOffset.
+      // position + buffer.remaining() > sizeInBytes.
       long readOverFlowCount = metrics.overflowReadError.getCount();
       try {
-        segment.readInto(readBuf, writeStartOffset + 1);
-        fail("Should have failed to read because position + buffer.remaining() > endOffset");
+        segment.readInto(readBuf, readBuf.remaining() + 1);
```

Comment:
I don't get this - could you explain why the position in the log segment from which we have to read `readBuf.remaining() + 1` and how it tests the case where `position + buffer.remaining() > sizeInBytes()` (i.e. they are not either independently greater than `sizeInBytes()` - only their sum is)?

I propose that the test be something like this:
```
readBuf = ByteBuffer.allocate(1);
segment.setEndOffset(segment.getStartOffset());
segment.readInto(readBuf, segment.sizeInBytes() - 1); 
// should succeed and assert that it equal to the last byte expected
readBuf = ByteBuffer.allocate(2);
segment.readInto(readBuf, segment.sizeInBytes() - 1); 
// should fail because position + buffer.remaining() > sizeInBytes.
```
This way we test that we can read until `sizeInBytes()` even if the `endOffset` is lower and that we aren't able to read past `sizeInBytes()`.